test/scripts/integration-tests.sh

```
@@ -0,0 +1,307 @@
+        echo "❌ moclojer server failed to start. Check logs:" >&2
+        cat "$LOG_FILE" >&2
+        exit 1
+    fi
+
+    # Check if server is responding
+    MAX_RETRIES=5
+    RETRY_COUNT=0
+    SERVER_READY=false
+
+    while [ $RETRY_COUNT -lt $MAX_RETRIES ] && [ "$SERVER_READY" = false ]; do
+        if curl -s -o /dev/null -w "%{http_code}" --max-time $TIMEOUT "http://$SERVER_HOST:$SERVER_PORT/hello/testuser" | grep -q "200"; then
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)